validation/views.py

```diff
@@ -381,6 +381,64 @@ def make_absolute_uri_for_speaker(code: str) -> str:
     return add_cors_headers(response)
 
 
+def bulk_search_recordings(request):
+    """
+    API endpoint to retrieve EXACT wordforms and return the URLs and metadata for the recordings.
+    Example: /api/bulk_search?q=mistik&q=minahik&q=waskay&q=mîtos&q=mistikow&q=mêstan
+    """
+
+    def make_absolute_uri_for_recording(rec: Recording) -> str:
+        uri = rec.compressed_audio.url
+        if uri.startswith("/"):
+            # It's a relative URI: build an absolute URI:
+            return request.build_absolute_uri(uri)
+
+        # It's an absolute URI already:
+        assert uri.startswith("http")
+        return uri
+
+    def make_absolute_uri_for_speaker(code: str) -> str:
+        return f"https://www.altlab.dev/maskwacis/Speakers/{code}.html"
+
+    query_terms = request.GET.getlist("q")
+    matched_recordings = []
+    not_found = []
+
+    for term in query_terms:
+        # Assume the query is an SRO transcription; prepare it for a fuzzy match.
+        result_set = Recording.objects.filter(
+            phrase__transcription=term,
+            speaker__gender__isnull=False,
+        )
+        # No bad recordings!
+        result_set = result_set.exclude(quality=Recording.BAD)
+        result_set = result_set.exclude(wrong_word=True)
+        result_set = result_set.exclude(wrong_speaker=True)
```

Comment:
Totally optional, but if you do `result_set = Recordings.objects.filter(…).exclude(quality=Recording.BAD).exclude(wrong_word=True).exclude(wrong_speaker=True)` and let black format it, you’ll be able to say `result_set` six fewer times.